cmd/gateway/main.go

```diff
@@ -25,7 +25,8 @@ func main() {
 	wsMux := mux.NewRouter()
 	cm := c.NewConnectionManager()
 	kw := queue.StartProducer(queue.GetProducer())
-	rc := ws.NewReceptorController(cm, wsMux, kw)
+	d := c.NewResponseDispatcherFactory(kw)
+	rc := ws.NewReceptorController(cm, wsMux, kw, d)
```

Comment:
Can we avoid passing the KafkaWriter instance (kw) to the ReceptorController constructor now?

Reply:
Good catch yeah we can avoid that. Will push an update